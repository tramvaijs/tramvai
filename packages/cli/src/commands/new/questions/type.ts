import any from '@tinkoff/utils/array/any';

type Type = 'app' | 'child-app';

const choices = [
  {
    name: 'Standalone tramvai app',
    value: 'app',
  },
  {
    name: 'Microfrontend implemented by tramvai child-app',
    value: 'child-app',
  },
];

const typeQuestion = (answer: string) => ({
  type: 'list' as const,
  name: 'type' as const,
  message: 'Choose a type for your project',
  choices,
  when: () => !any<{ name: string; value: string }>((choice) => choice.value === answer)(choices),
  default: 'app' as const,
});

export { Type, typeQuestion };
