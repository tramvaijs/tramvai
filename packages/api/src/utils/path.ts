import path from 'node:path';

export const resolveAbsolutePathForFile = ({
  file,
  sourceDir,
  rootDir,
}: {
  file: string;
  sourceDir: string;
  rootDir: string;
}): string => {
  if (path.isAbsolute(file)) {
    return file;
  }
  if (path.isAbsolute(sourceDir)) {
    return path.join(sourceDir, file);
  }
  if (path.isAbsolute(rootDir)) {
    return path.join(rootDir, sourceDir, file);
  }
  return path.resolve(rootDir, sourceDir, file);
};

export const resolveAbsolutePathForFolder = ({
  folder,
  rootDir,
}: {
  folder: string;
  rootDir: string;
}): string => {
  if (path.isAbsolute(folder)) {
    return folder;
  }
  if (path.isAbsolute(rootDir)) {
    return path.join(rootDir, folder);
  }
  return path.resolve(rootDir, folder);
};
