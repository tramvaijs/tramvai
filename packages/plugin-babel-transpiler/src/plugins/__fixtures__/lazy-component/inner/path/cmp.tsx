import React from 'react';

const InnerCmp = () => {
  return <div>inner cmp</div>;
}

export default InnerCmp;
