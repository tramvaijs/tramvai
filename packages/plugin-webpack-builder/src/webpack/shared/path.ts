import path from 'node:path';

export const resolveAbsolutePathForEntry = ({
  entry,
  sourceDir,
  rootDir,
}: {
  entry: string;
  sourceDir: string;
  rootDir: string;
}): string => {
  if (path.isAbsolute(entry)) {
    return entry;
  }
  if (path.isAbsolute(sourceDir)) {
    return path.join(sourceDir, entry);
  }
  if (path.isAbsolute(rootDir)) {
    return path.join(rootDir, sourceDir, entry);
  }
  return path.resolve(rootDir, sourceDir, entry);
};

export const resolveAbsolutePathForOutput = ({
  output,
  rootDir,
}: {
  output: string;
  rootDir: string;
}): string => {
  if (path.isAbsolute(output)) {
    return output;
  }
  if (path.isAbsolute(rootDir)) {
    return path.join(rootDir, output);
  }
  return path.resolve(rootDir, output);
};
