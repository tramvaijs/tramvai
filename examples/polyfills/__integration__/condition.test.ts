import path from 'path';
import type { StartCliResult } from '@tramvai/test-integration';
import { startCli } from '@tramvai/test-integration';
import { initPlaywright } from '@tramvai/test-pw';

jest.setTimeout(30000);

describe('polyfill condition', () => {
  let app: StartCliResult;

  beforeAll(async () => {
    app = await startCli('polyfills', {
      rootDir: path.resolve(__dirname, '../'),
      fileCache: false,
    });
  }, 80000);

  afterAll(() => {
    return app.close();
  });

  it('polyfill condition exists in html', async () => {
    const { browser, getPageWrapper } = await initPlaywright(app.serverUrl);
    const { page } = await getPageWrapper(app.serverUrl);

    const polyfillConditionScriptContent = await page.$eval(
      '#polyfills',
      (node) => (node as HTMLElement).innerHTML
    );

    expect(
      polyfillConditionScriptContent.replace(app.staticUrl, 'http://localhost:4000')
    ).toMatchSnapshot();

    await browser.close();
  });
});
