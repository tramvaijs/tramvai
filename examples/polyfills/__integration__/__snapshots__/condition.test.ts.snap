// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`polyfill condition polyfill condition exists in html 1`] = `
"(function (){
  var con;
  try {
    con = !(Object.hasOwn) || !('at' in window.Array.prototype) || false;
  } catch (e) {
    con = true;
  }
  if (con) { window.TRAMVAI_POLLYFILL_LOADED = true;
  document.write('<script defer="defer" charset="utf-8" data-critical="true" crossorigin="anonymous" src="http://localhost:4000/dist/client/polyfill.js"><\\/script>')}
})()"
`;
